cli/cli.go

```diff
@@ -0,0 +1,111 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	data "github.com/tendermint/go-wire/data"
+	"github.com/tendermint/go-wire/data/base58"
+)
+
+const (
+	RootFlag     = "root"
+	OutputFlag   = "output"
+	EncodingFlag = "encoding"
+)
+
+func PrepareMainCmd(cmd *cobra.Command, envPrefix, defautRoot string) func() {
```

Comment:
Can we have a couple sentences at the top of this command describing a usage scenario? Also add option for ExecuteWithDebug? (see PR from `basecoin-cli`)